lib/rucio/gateway/account_limit.py

```
@@ -47,110 +47,74 @@ def get_rse_account_usage(
 ) -> dict[str, Union[int, float, None]]:
     """
-    Lists the limitation names/values for the specified account name and rse name.
+    Lists the limitation names/values for the specified account name.
+    If an RSE is provided, it returns the limit for that specific RSE.
+    Otherwise, it returns all account limits.
 
     REST API: http://<host>:<port>/rucio/account/<account>/limits
 
-    :param account:     The account name.
-    :param rse:         The rse name.
-    :param vo:          The VO to act on.
-    :param session:     The database session in use.
+    :param account: The account name.
+    :param rse: The RSE name (optional).
+    :param vo: The VO to act on.
+    :param session: The database session in use.
 
-    :returns: The account limit.
+    :returns: A dictionary of account limits with RSE names as keys and limits as values.
     """
 
     internal_account = InternalAccount(account, vo=vo)
 
-    rse_id = get_rse_id(rse=rse, vo=vo, session=session)
-    return {rse: account_limit_core.get_local_account_limit(account=internal_account, rse_id=rse_id, session=session)}
-
-
-@read_session
-def get_global_account_limits(
-    account: str,
-    vo: str = 'def',
-    *,
-    session: "Session"
-) -> dict[str, RSEResolvedGlobalAccountLimitDict]:
-    """
-    Lists the limitation names/values for the specified account name.
-
-    REST API: http://<host>:<port>/rucio/account/<account>/limits
-
-    :param account:     The account name.
-    :param vo:          The VO to act on.
-    :param session:     The database session in use.
-
-    :returns: The account limits.
-    """
-    if account:
-        internal_account = InternalAccount(account, vo=vo)
+    if rse:
+        # Single RSE lookup
+        rse_id = get_rse_id(rse=rse, vo=vo, session=session)
+        return {rse: account_limit_core.get_local_account_limit(account=internal_account, rse_ids=rse_id, session=session)}
     else:
-        internal_account = InternalAccount('*', vo=vo)
```

This functionality is lost in the new version. I'm not sure how important it is, so maybe someone with more insight here can chime in. I would keep it just to be safe.

---

It didn't made sense to me since `account` is not an optional param.

---

It matters in the context of the [`if/else` it's found in](
https://github.com/rucio/rucio/blob/34a85aa511aaca69d59fa422b3f6a201375855db/lib/rucio/gateway/account_limit.py#L122C1-L125C55):

```
    if account:
        internal_account = InternalAccount(account, vo=vo)
    else:
        internal_account = InternalAccount('*', vo=vo)
```

`account` could be `''`, in which case it would evaluate to `False`.

```
>>> account = ''
>>> bool(account)
False
```

In that case, it would be set as `'*'`.

---

Should an empty string `' '` be considered a valid value for account? If not, maybe we should raise an error instead of assigning '*'.
We can also make the condition clearer by explicitly checking for None:
```
if account is not None:  
    internal_account = InternalAccount(account, vo=vo)```

---

> Should an empty string ' ' be considered a valid value for account?

I assume it shouldn't, but the SQLA model actually doesn't seem to check against that. I agree with you that it would be clearer if we checked `is not None` instead, and then somewhere in the call (I guess in core) we have the logic that actually finds all account limits if a single `account` is not specified.

I think it's a good idea if someone that frequently uses this `account_limit` feature could chime in on what the use case is for having this kind of logic. Maybe you could ask in either the support or developers channel on Mattermost